development/kyma-github-connector/githubWebhookGateway/pkg/gateway/webhookhandler.go

```diff
@@ -52,20 +71,45 @@ func (wh *WebHookHandler) HandleWebhook(w http.ResponseWriter, r *http.Request)
 		httperrors.SendErrorResponse(apperr, w)
 		return
 	}
-	var eventType string
+
+	// Check if event is supported by a code and allowed for this instance.
 	switch event := event.(type) {
 	// Supported github events
 	case *github.IssuesEvent:
-		eventType = fmt.Sprintf("issuesevent.%s", *event.Action)
+		eventGroup := "issuesevent"
+		eventType, supported = wh.checkIfEventSupported(allowedEvents, eventGroup, *event.Action)
+	case *github.PullRequestEvent:
+		eventGroup := "pullrequest"
+		eventType, supported = wh.checkIfEventSupported(allowedEvents, eventGroup, *event.Action)
+	default:
+		supported = false
 	}
 
-	sourceID := os.Getenv("GITHUB_WEBHOOK_GATEWAY_NAME")
-	log.Info(fmt.Sprintf("received event of type: %s", eventType))
-	apperr = wh.sender.SendToKyma(eventType, sourceID, payload)
+	if supported {
+		// CloudEvents sourceID.
+		sourceID := os.Getenv("GITHUB_WEBHOOK_GATEWAY_NAME")
+		log.Info(fmt.Sprintf("received event of type: %s", eventType))
+		apperr = wh.sender.SendToKyma(eventType, sourceID, payload)
 
-	if apperr != nil {
-		log.Info(apperrors.Internal("while handling the event: %s", apperr.Error()))
-		return
+		if apperr != nil {
```

Comment:
`apperr` -> `err`
No need to name an error that way. This name got me confused at first place.